libmamba/include/mamba/core/history.hpp

```diff
@@ -61,6 +62,22 @@ namespace mamba
         ChannelContext& m_channel_context;
     };
 
+    namespace detail
+    {
+        /** PackageDiff contains two maps of packages and their package info, one being for the
+         * installed packagege, the other for the removed ones. This is used while looping on
+         * revisions to get the diff between the target revision and the current one. */
+        struct PackageDiff
+        {
+            std::unordered_map<std::string, specs::PackageInfo> removed_pkg_diff;
+            std::unordered_map<std::string, specs::PackageInfo> installed_pkg_diff;
+
+            PackageDiff
+            get_revision_pkg_diff(std::vector<History::UserRequest> user_requests, int target_revision);
```

Comment:
```suggestion
            get_revision_pkg_diff(std::vector<History::UserRequest> user_requests, std::size_t target_revision);
```

Reply:
Also is this supposed to be `static`, i.e. do you access both the current `PackageDiff` *and* the one you are creating?